src/main/java/com/puppycrawl/tools/checkstyle/checks/header/HeaderCheck.java

```diff
@@ -88,13 +89,13 @@ public void setIgnoreLines(int... list) {
     }
 
     @Override
-    protected void processFiltered(File file, List<String> lines) {
-        if (getHeaderLines().size() > lines.size()) {
+    protected void processFiltered(File file, FileText fileText) {
+        if (getHeaderLines().size() > fileText.getLines().size()) {
             log(1, MSG_MISSING);
         }
         else {
             for (int i = 0; i < getHeaderLines().size(); i++) {
-                if (!isMatch(i, lines.get(i))) {
+                if (!isMatch(i, fileText.getLines().get(i))) {
```

Comment:
calling `getLines` in a loop looks very bad. We are creating new instances each loop.
Please make a temporary variable before entering the loop.
You could use temporary variable for previous call to `getLines` above a few lines.